node/debug.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"net"
+
+	"github.com/tendermint/tendermint/config"
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/service"
+	rpccore "github.com/tendermint/tendermint/rpc/core"
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/store"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Debug is a type useful for debugging tendermint problems.
```

Comment:
I would call Debug a node mode for debugging tendermint problems

Reply:
👍🏼 for being more explicit about what exactly this type is—is it a node itself? A service? As a potential reader of this comment, not knowing the context, I'd like to come away from this comment knowing why I would create a value of this type. 

The following is an attempt based on incomplete understanding and the existing wording, just to illustrate what I mean:

```go
// A Debug value manages an RPC service that exports methods to debug a failed node.
// After a node shuts down due to consensus failure, it will no longer start up and cannot
// easily be inspected. A Debug value provides a similar interface to the node, using the
// underlying Tendermint data stores, without bringing up any other components.
// A caller can query the Debug service to inspect the persisted state and debug the failure.
```

or words to that effect (what I wrote may well be wrong).

Reply:
Grabbed what @creachadair put here. Agreed about having the comment describe when the node should be used.